kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -519,34 +812,52 @@ This unification case throws an error because domain values may not occur in a
 sort with constructors.
 -}
 domainValueAndConstructorErrors ::
-    Monad unifier =>
     HasCallStack =>
     TermLike RewritingVariableName ->
     TermLike RewritingVariableName ->
-    MaybeT unifier a
-domainValueAndConstructorErrors
-    term1@(DV_ _ _)
-    term2@(App_ secondHead _)
-        | Symbol.isConstructor secondHead =
-            error
-                ( unlines
-                    [ "Cannot handle DomainValue and Constructor:"
-                    , unparseToString term1
-                    , unparseToString term2
-                    ]
-                )
-domainValueAndConstructorErrors
-    term1@(App_ firstHead _)
-    term2@(DV_ _ _)
-        | Symbol.isConstructor firstHead =
-            error
-                ( unlines
-                    [ "Cannot handle Constructor and DomainValue:"
-                    , unparseToString term1
-                    , unparseToString term2
-                    ]
-                )
-domainValueAndConstructorErrors _ _ = empty
+    DVConstrError ->
+    unifier a
+domainValueAndConstructorErrors term1 term2 unifyData =
+    error
+        ( unlines
+            [ cannotHandle
+            , unparseToString term1
+            , unparseToString term2
+            ]
+        )
```

Comment:
I know this is not your change, but for consistency, we should probably format this error with `Pretty`.